Issue-253 Configure cache so EhCache is used instead of the default cache

wmvsilva opened the pull request:
See #253.

The cache was not set up correctly and was using the default simple Spring implementation instead of the config in `ehcache.xml`. This PR fixes this so the `ehcache.xml` config is used.

Commit:
Issue-253 Configure cache so EhCache is used instead of the default cache

wmvsilva committed:
Merge pull request #260 from wmvsilva/feature/issue-253_fix-cache-config

Issue-253 Configure cache so EhCache is used instead of the default cache

GalateaEric commented on src/main/java/org/galatea/starter/AppConfig.java:
```diff
@@ -4,17 +4,25 @@
 public class AppConfig {
 
+  @Value("${cache-config}")
+  public String cacheConfig;
```
can we move this to not be a member variable, and instead do an @Value on a method parameter to the @Bean ehCacheCacheManager() method

wmvsilva replied:
Fixed

GalateaEric reviewed (commented):
Can we create some unit tests that demonstrate the EhCache usage -- some what as you described in the test evidence of the issue.  To guard against someone accidentally disabling EhCache with a change in the future.

wmvsilva committed:
Merge branch 'develop' into feature/issue-253_fix-cache-config

wmvsilva committed:
issue-253 Add EhCache test & move @Value cache-config as param

wmvsilva commented:
I've made changes based on @GalateaEric's comments. I've added a Unit Test to confirm that EhCache can be properly created/configured from an ehcache.xml config file.

It would be nice to have more extensive testing that the Cache is actually caching missions for the settlements endpoint. However, it looks like that would involve expanding the tests to run a full version of the app. While we already have integration tests, those actually connect to https://fuse-rest-dev.cfapps.io to perform the tests. In order to test specific behavior of a cache that we can easily configure, I think that a full instance of the app with an H2 database should be brought up as part of the normal tests. I've done this in other API projects before where the `application.yml` is fully populated for tests and connects to a local database that we populate before test queries are run.

cbaltera committed:
Merge branch 'develop' into feature/issue-253_fix-cache-config

cbaltera commented on src/main/java/org/galatea/starter/AppConfig.java:
```diff
@@ -38,6 +43,29 @@ public IAgreementTransformer agreementTransformer() {
+   * @return factory bean for the EhCache to be passed to EhCacheCacheManager.
+   */
+  @Bean
+  public EhCacheManagerFactoryBean ehCacheCacheManager(
```
Can we change the name of this bean so that it's clear that it isn't itself the cache manager? (and also its use as param in the bean above)

wmvsilva replied:
I've updated the name of the bean and the parameter in the bean above.

Commit:
Issue-253 Change name of cache manager factory bean

Changes the name of the cache manager factory bean to reflect that it is actually a cache manager factory bean and not a cache manager

Commit:
Merge branch 'develop' into feature/issue-253_fix-cache-config
